draft-ietf-taps-interface.md

```diff
@@ -613,10 +613,111 @@ Preconnection during pre-establishment.
-endpoint identifiers, such as IP addresses, DNS names, and interface names,
-as well as port numbers and service names.
+to refer to the endpoints of a transport connection. Endpoints can be created
+as either Remote or Local:
+
+~~~
+RemoteSpecifier := NewRemoteEndpoint()
+LocalSpecifier := NewLocalEndpoint()
+~~~
+
+A single Endpoint Object represents the identity of a network host. That endpoint
+can be more or less specific depending on which identifiers are set. For example,
+an Endpoint that only specifies a hostname may in fact end up corresponding
+to several different IP addresses on different hosts.
+
+An Endpoint Object can be configured with the following identifiers:
+
+- Hostname (string)
+
+~~~
+RemoteSpecifier.WithHostname("example.com")
+~~~
+
+- Port (a 16-bit integer) or a Service (string) that maps to a port
+
+~~~
+RemoteSpecifier.WithService(443)
+~~~
+
+~~~
+RemoteSpecifier.WithService("https")
+~~~
+
+- IP address (IPv4 or IPv6 address)
+
+~~~
+RemoteSpecifier.WithIPv4Address(192.0.2.21)
+~~~
+
+~~~
+RemoteSpecifier.WithIPv6Address(2001:db8:4920:e29d:a420:7461:7073:0a)
+~~~
+
+- Interface (string name)
+
+~~~
+LocalSpecifier.WithInterface("en0")
+~~~
+
+- Protocol (implementation-specific enumeration)
+
+~~~
+LocalSpecifier.WithProtocol(TCP)
+~~~
+
+An Endpoint cannot have multiple identifiers of a same type set. That is,
+an endpoint cannot have two IP addresses specified. Two separate IP addresses
+are represented as two Endpoint Objects. If a Preconnection specifies a Remote
+Endpoint with a specific IP address set, it will only establish Connections to
+that IP address. If, on the other hand, the Remote Endpoint specifies a hostname
+but no addresses, the Connection can perform name resolution and attempt
+using any address derived from the original hostname Remote Endpoint.
+
+The Transport Services API resolves names internally, when the Initiate(),
+Listen(), or Rendezvous() method is called to establish a Connection. Privacy
+considerations for the timing of this resolution are given in {{privacy-security}}.
+
+The Resolve() action on Preconnection can be used by the application to force
+early binding when required, for example with some Network Address Translator
+(NAT) traversal protocols (see {{rendezvous}}).
+
+### Using Multicast Endpoints
+
+Specifying a multicast group address on a Local Endpoint will indicate to the transport
+system that the resulting connection will be used to receive multicast messages. The
+Remote Endpoint can be used to filter incoming multicast from specific senders. Such
+a Preconnection will only support calling Listen(), not Initiate(). The accepted Connections
+are receive-only.
+
+Similarly, specifying a multicast group address on the Remote Endpoint will indicate that the
+resulting connection will be used to send multicast messages.
+
+### Endpoint Aliases
+
+An Endpoint can have an alternative definition when using different protocols.
```

Comment:
I'm just wondering: is this limited to different protocols? Or would aliases also be a way to, e.g., bundle different IP addresses together to one "alias" ?

IF it is strictly about protocols, maybe the wording could be a bit more explicit about this being the only use case. Maybe insert a limiting sentence here... like "Aliases cannot be used to specify ... " ?  These are just vague thoughts, do whatever you want with them. Totally fine to ignore all of this.

Reply:
I'm not sure if we want to limit the alias to this use case only, so let's leave it a bit loose for now, and discuss more?

Reply:
OK!  I like loose. But then, why not say something like "can have an alternative definition with different identifiers." ?   Else, "when using different protocols" really sounds like the only possible case.  Just a thought - if you *want* it half-loose as it is, that's also ok with me.